python/lsst/pipe/analysis/plotUtils.py

```diff
@@ -356,17 +362,26 @@ def plotCcdOutline(axes, butler, dataId, ccdList, zpLabel=None, fontSize=8):
 
         ras = list()
         decs = list()
+        coords = list()
         for x, y in zip([0, w, w, 0, 0], [0, 0, h, h, 0]):
             xy = afwGeom.Point2D(x, y)
             ra = np.rad2deg(np.float64(wcs.pixelToSky(xy)[0]))
             dec = np.rad2deg(np.float64(wcs.pixelToSky(xy)[1]))
             ras.append(ra)
             decs.append(dec)
-        axes.plot(ras, decs, "k-", linewidth=1)
+            coords.append(afwGeom.SpherePoint(ra, dec, afwGeom.degrees))
         xy = afwGeom.Point2D(w/2, h/2)
         centerX = np.rad2deg(np.float64(wcs.pixelToSky(xy)[0]))
         centerY = np.rad2deg(np.float64(wcs.pixelToSky(xy)[1]))
-        axes.text(centerX, centerY, "%s" % str(ccdLabelStr), ha="center", va="center", fontsize=fontSize)
+        inTract = False
+        if tractInfo is not None:
+            for coord in coords:
+                if tractInfo.contains(coord):
+                    inTract = True
+                    break
+        if not tractInfo or inTract:
```

Comment:
This checks if tractInfo is not None, or an empty sequence or something else that evaluates to False

Reply:
Yeah...it's ugly, but the "inTract" check is to ward against plotting empty outlines for ccds not in the tract. `jointcal` persists output for them all, but I'm only reading in catalogs for those overlapping the tract...I can't guarantee the appropriate CCD list for the comparison between `meas_mosaic` and `jointcal` as they will differ...hopefully this can also get fixed up when the former is abandoned.  The tractInfo check is to allow for all CCDs to be plotted regardless of tract overlap (e.g. I pass in None for the "commonZP" cats as these do not have uberCals applied, so there's no reason not to plot the entire visit).